test/conformance/conformance_flowgraph.h

```diff
@@ -17,74 +17,810 @@
+        static bool check = false;
+        if(check) {
+            check = false;
+            fc.stop();
+            return OutputType(1);
+        }
+        check = true;
+        return OutputType(1);
+    }
+};
+
+struct wait_flag_body {
+    static std::atomic<bool> flag;
+
+    wait_flag_body() {
+        flag.store(false);
+    }
+
+    template<typename InputType>
+    InputType operator()( InputType ) {
+        while(!flag.load()) { utils::yield(); };
+#ifdef CONFORMANCE_CONTINUE_NODE
+        return InputType();
+#else
+        return InputType(0);
+#endif
+    }
+
+    template<typename InputType>
+    void operator()( InputType argument, multifunc_ports_t<InputType>& op ) {
+        while(!flag.load()) { };
+        std::get<0>(op).try_put(argument);
+    }
+
+    template<typename InputType>
+    void operator()( InputType argument, assync_ports_t<InputType>& g ) {
+        while(!flag.load()) { };
+        g.try_put(argument);
+    }
+};
+
+std::atomic<bool> wait_flag_body::flag{false};
+
+struct concurrency_peak_checker_body {
+    std::size_t required_max_concurrency = 0;
+
+    concurrency_peak_checker_body( std::size_t req_max_concurrency = 0 ) :
+                                    required_max_concurrency(req_max_concurrency) {}
+
+    concurrency_peak_checker_body( const concurrency_peak_checker_body & ) = default;
+
+    int operator()( oneapi::tbb::flow_control & fc ) {
+        static int counter = 0;
+        utils::ConcurrencyTracker ct;
+        if(++counter > 500) {
+            counter = 0;
+            fc.stop();
+            return 1;
+        }
+        utils::doDummyWork(1000);
+        CHECK_MESSAGE((int)utils::ConcurrencyTracker::PeakParallelism() <= required_max_concurrency,
+        "Input node is serial and its body never invoked concurrently");
+        return 1;
+    }
+
+    int operator()( int ) {
+        utils::ConcurrencyTracker ct;
+        utils::doDummyWork(1000);
+        CHECK_MESSAGE((int)utils::ConcurrencyTracker::PeakParallelism() <= required_max_concurrency,
+        "Measured parallelism is not expected");
+        return 1;
+    }
+
+    void operator()( const int& argument, multifunc_ports_t<int>& op ) {
+        utils::ConcurrencyTracker ct;
+        utils::doDummyWork(1000);
+        CHECK_MESSAGE((int)utils::ConcurrencyTracker::PeakParallelism() <= required_max_concurrency,
+        "Measured parallelism is not expected");
+        std::get<0>(op).try_put(argument);
+    }
+
+    void operator()( const int& argument , assync_ports_t<int>& g ) {
+        utils::ConcurrencyTracker ct;
+        utils::doDummyWork(1000);
+        CHECK_MESSAGE((int)utils::ConcurrencyTracker::PeakParallelism() <= required_max_concurrency,
+        "Measured parallelism is not expected");
+        g.try_put(argument);
+    }
+};
+
+template<typename OutputType, typename InputType = int>
+struct counting_object {
+    std::size_t copy_count;
+    mutable std::size_t copies_count;
```

Comment:
Not sure I understand why there is two different variables with almost similar name. Could you please explain?
Also, what prevents us from making them class variables?